File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    queue = deque()
    
    for i in range(len(progresses)):
        day = (100 - progresses[i] + speeds[i] - 1) // speeds[i]
        queue.append(day)
    
    cur_day = queue.popleft()
    count = 1
    
    while queue:
        day = queue.popleft()
        if day <= cur_day:
            count += 1
        else:
            answer.append(count)
            cur_day = day
            count = 1
    
    answer.append(count)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> q;
    int n = progresses.size();
    
    for (int i=0; i<n; i++) {
        int progress = progresses[i];
        int speed = speeds[i];
        q.push((100 - progress - 1) / speed + 1);
    }
    
    int day = q.front();
    q.pop();
    int temp = 1;
    
    while (q.size() > 0) {
        int cur_day = q.front();
        q.pop();
        if (day >= cur_day) {
            temp += 1;
        } else {
            answer.push_back(temp);
            day = cur_day;
            temp = 1;
        }
    }
    
    answer.push_back(temp);
    
    return answer;
}